python-package/xgboost/spark/data.py

```diff
@@ -149,24 +153,52 @@ def _read_csr_matrix_from_unwrapped_spark_vec(part: pd.DataFrame) -> csr_matrix:
     )
 
 
+def make_qdm(
+    data: Dict[str, List[np.ndarray]],
+    gpu_id: Optional[int],
+    meta: Dict[str, Any],
+    ref: Optional[DMatrix],
+    params: Dict[str, Any],
+) -> DMatrix:
+    """Handle empty partition for QuantileDMatrix."""
+    if not data:
+        return QuantileDMatrix(np.empty((0, 0)), ref=ref)
+    it = PartIter(data, gpu_id, **meta)
+    m = QuantileDMatrix(it, **params, ref=ref)
+    return m
+
+
 def create_dmatrix_from_partitions(  # pylint: disable=too-many-arguments
     iterator: Iterator[pd.DataFrame],
     feature_cols: Optional[Sequence[str]],
     gpu_id: Optional[int],
+    use_qdm: bool,
     kwargs: Dict[str, Any],  # use dict to make sure this parameter is passed.
     enable_sparse_data_optim: bool,
     has_validation_col: bool,
 ) -> Tuple[DMatrix, Optional[DMatrix]]:
-    """Create DMatrix from spark data partitions. This is not particularly efficient as
-    we need to convert the pandas series format to numpy then concatenate all the data.
+    """Create DMatrix from spark data partitions.
 
     Parameters
     ----------
     iterator :
         Pyspark partition iterator.
+    feature_cols:
+        A sequence of feqture names, used only when rapids plugin is enabled.
```

Comment:
feqture -> feature. this parameter can be used even without rapid plugin.

Reply:
I haven't really done any test for it and it will likely trigger an assert error. We have `DMatrix` and `QuantileDMatrix` to support, I will leave that to the next release.